agr_java_core/src/main/java/org/alliancegenome/es/index/site/schema/Mapping.java

```diff
@@ -1,11 +1,47 @@
 package org.alliancegenome.es.index.site.schema;
 
-import java.io.IOException;
-
+import lombok.SneakyThrows;
+import org.alliancegenome.curation_api.model.document.es.AffectedGenomicModelDocument;
+import org.apache.commons.lang3.StringUtils;
 import org.elasticsearch.xcontent.XContentBuilder;
 
+import java.io.IOException;
+
 public class Mapping extends Builder {
 
+	/**
+	 * Converts a method name to a field name by stripping the "get" prefix and lowercasing the first letter.
+	 * For example, "getGeneSymbol" becomes "geneSymbol".
+	 *
+	 * @param clazz      The class containing the method.
+	 * @param methodName The method name to convert.
+	 * @return The mapped field name.
+	 */
+	public static String getMappedFieldNameByMethodName(Class clazz, String methodName) {
+
+		try {
+			String methodNameString = clazz.getMethod(methodName).getName();
+			// strip off the "get" or "is" prefix
+			String[] token = StringUtils.splitByCharacterTypeCamelCase(methodNameString);
+			String strippedOffGet = methodNameString.replace(methodName, token[0]);
+
+			// lower case the first letter
+			String fieldName = methodNameString.substring(strippedOffGet.length());
+			String output = fieldName.substring(0, 1).toLowerCase() + fieldName.substring(1);
+			return output;
+		} catch (NoSuchMethodException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@SneakyThrows
+	public static String getMappedFieldNameByMethodName(String classAndMethodName) {
```

Comment:
I find this code very confusing... can we just hard code the names of the fields below, then it is explicit of which fields are getting indexed.